fourm/data/modality_transforms.py

```diff
@@ -706,13 +748,51 @@ def image_augment(
                 "Crop settings / augmentation index are missing but a pre-tokenized modality is being used"
             )
         v = torch.tensor(v[rand_aug_idx])
-        return v # Since we augment before saving (create 5 augmented versions of the same image), in this transform you randomly sample one of the augmentations to work with. 
+        return v  # Since we augment before saving (create 5 augmented versions of the same image), in this transform you randomly sample one of the augmentations to work with.
 
     def postprocess(self, sample):
         return sample
 
 
-class DetectionTransform(AbstractTransform): # bounding boxes
+class VideoTokTransform(AbstractTransform):
+    """
+    Assume input tokens is an ndarray of shape (num_frames, num_tokens_per_frame).
+    Transform the tokens to a torch tensor of shape (num_frames * num_tokens,).
+    """
+    # Transformation on the tokens
+
+    def __init__(self):
+        pass
+
+    def load(self, path):
+        sample = np.load(path).astype(int)
+        return sample  # shape: (num_frames, num_tokens_per_frame)
+
+    def preprocess(self, sample):
+        return sample
+
+    def image_augment(
```

Comment:
Role of this: do the augmentation during the tokenization step here

Reply:
TokTransform's v input is actually a list of tokenized inputs which are different augmentations of the same input. So that transform requires an index to select one of the augmented inputs.

Reply:
So here, we should assume no augmentations but keep the input v as a list of augmentations. but in this case since there's none just get v[0] always.